src/main/java/org/jenkinsci/plugins/xunit/pipeline/XUnitResultsStepExecution.java

```diff
@@ -0,0 +1,63 @@
+package org.jenkinsci.plugins.xunit.pipeline;
+
+import hudson.AbortException;
+import hudson.FilePath;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.junit.TestResultAction;
+import hudson.tasks.junit.TestResultSummary;
+import hudson.tasks.junit.pipeline.JUnitResultsStepExecution;
+import org.jenkinsci.lib.dtkit.type.TestType;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.SynchronousStepExecution;
+import org.jenkinsci.plugins.xunit.ExtraConfiguration;
+import org.jenkinsci.plugins.xunit.XUnitProcessor;
+import org.jenkinsci.plugins.xunit.threshold.XUnitThreshold;
+
+import javax.annotation.Nonnull;
+
+public class XUnitResultsStepExecution extends SynchronousStepExecution<TestResultSummary> {
```

Comment:
Prefer `SynchronousNonBlockingStepExecution`, especially as you are doing Remoting operations.

Reply:
Well you can add an explicit “running” flag or something. But you should still use `SynchronousNonBlockingStepExecution`.

Reply:
I agree with @jglick here -- this really should not be blocking, because it's potentially a longer-running operation and that means your thread can't satisfy any other calls (including potentially some UI operations as with InputStepExecution, sigh).

You should be able to do some internal synchronization magic to let it politely wait for the other junit execution before doing meaningful work, without blocking the CPS thread.

Reply:
Check my latest push - might be what you're looking for?